### PR TITLE
Moving last few image operations to engine-api

### PR DESCRIPTION
**Pull request by nishanttotla**
Addresses #1879

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

### CODE REVIEW

**Comment by dongluochen** [comment]
LGTM

File: cluster/engine.go [hunk 1]
```diff
@@ -1015,12 +1015,28 @@ func (e *Engine) Pull(image string, authConfig *types.AuthConfig) error {
 
 // Load an image on the engine
 func (e *Engine) Load(reader io.Reader) error {
-	err := e.client.LoadImage(reader)
+	loadResponse, err := e.apiClient.ImageLoad(context.TODO(), reader, false)
```
Comment (abronan) [inline cluster/engine.go]: Hmm can we replace those `context.TODO()` with a real context here...

File: cluster/engine.go [hunk 2]
```diff
@@ -1029,7 +1045,13 @@ func (e *Engine) Load(reader io.Reader) error {
...
 // Import image
 func (e *Engine) Import(source string, repository string, tag string, imageReader io.Reader) error {
-	_, err := e.client.ImportImage(source, repository, tag, imageReader)
+	opts := types.ImageImportOptions{
+		SourceName:     source,
+		RepositoryName: repository,
+		Tag:            tag,
+		Source:         imageReader,
+	}
+	_, err := e.apiClient.ImageImport(context.TODO(), opts)
```
Comment (abronan) [inline cluster/engine.go]: same here

File: cluster/engine.go [hunk 3]
```diff
@@ -1247,7 +1269,13 @@ func (e *Engine) BuildImage(buildImage *types.ImageBuildOptions) (io.ReadCloser,
...
 func (e *Engine) TagImage(IDOrName string, repo string, tag string, force bool) error {
 	// send tag request to docker engine
-	err := e.client.TagImage(IDOrName, repo, tag, force)
+	opts := types.ImageTagOptions{
+		ImageID:        IDOrName,
+		RepositoryName: repo,
+		Tag:            tag,
+		Force:          force,
+	}
+	err := e.apiClient.ImageTag(context.TODO(), opts)
```
Comment (abronan) [inline cluster/engine.go]: same

Comment (vieux) [inline cluster/engine.go]: todo is everywhere...

File: cluster/engine.go (hunk 1 above)
Comment (nishanttotla) [inline cluster/engine.go]: We've been using `context.TODO()` for all `engine-api` calls so far. What do you recommend?

Comment (abronan) [inline cluster/engine.go]: We can leave it as is for now until the integration with engine-api is over and we determine where the use of a context is critical, but we should use the contexts properly deriving a default context and adding timeouts when necessary or cancel the context under exceptional circumstances. There are no guarantee that the call to `ImageLoad` will ever return. And the empty context does not capture a timeout value.

**Comment by abronan** [comment]
LGTM

File: cluster/engine.go (hunk 1 above)
Comment (nishanttotla) [inline cluster/engine.go]: @abronan I agree. Let's tackle that soon.

**Pull request merged**